packages/kit/src/runtime/client/client.js

```diff
@@ -223,7 +223,9 @@ export function create_client({ target, base }) {
 
 	/** @param {...string} pathnames */
 	async function preload_code(...pathnames) {
-		const matching = routes.filter((route) => pathnames.some((pathname) => route.exec(pathname)));
+		const matching = routes.filter(
+			(route) => !pathnames.length || pathnames.some((pathname) => route.exec(pathname))
+		);
```

Comment:
```suggestion
		const matching = pathnames.length
			? routes.filter((route) => pathnames.some((pathname) => route.exec(pathname)))
			: routes;
```

To avoid running `.filter` unnecessarily, and to avoid checking `pathnames.length` for every route.

Reply:
I had this initially but this is a few characters shorter 😅 happy change if we think it's better (I don't think performance is a concern here, but your suggestion arguably more readable )

Reply:
After passing through Terser, they appear to be an identical number of bytes 😛

`e.filter((l=>!e.length||e.some((e=>l.exec(e)))))`
`e.length?t.filter((t=>e.some((e=>t.exec(e))))):t`